ADA/databricks_rstudio_sql_warehouse.qmd

```diff
@@ -1,105 +1,255 @@
+-   Modifying your .Renviron file to establish a connection between RStudio and Databricks
+-   Adding connection code to your existing scripts in RStudio
+
+Each of these steps is described in more detail in the sections below. 
+
+---
+
+### Setting up the ODBC driver
+
+---
+
+#### Install the 'Simba Spark ODBC' driver from the Software Centre
+
+---
+
+-   Open the Software Centre via the start menu
+
+-   In the 'Applications' tab, click 'Simba Spark ODBC Driver 64-bit'
+
+::: {align="center"}
+![](../images/databricks-software-centre.png)
+:::
+
+-   Click install
+
+------------------------------------------------------------------------
+
+#### Get connection details for the SQL Warehouse from Databricks
+
+------------------------------------------------------------------------
+
+To set up the connection you will need a few details from the SQL Warehouse within Databricks that you wish to connect to.
+
+-   Log in to Databricks
+
+-   Click on the 'SQL Warehouses' tab in the sidebar
+
+::: {align="center"}
+![](../images/databricks-SQL-warehouses.png)
+:::
+
+-   Click on the name of the warehouse you want to connect to, and click the 'Connection Details' tab
+
+-   Make a note of the 'Server hostname', 'Port', and 'HTTP Path'
 
 ------------------------------------------------------------------------
 
-# Pre-requisites
+#### Get a personal access token from Databricks for authentication
 
 ------------------------------------------------------------------------
 
-You must have:
+A personal access token is is a security measure that acts as an identifier to let Databricks know who is accessing information from the SQL warehouse. Access tokens are usually set for a limited amount of time, so they will need renewing periodically.
+
+-   In Databricks, click on your email address in the top right corner, then click 'User settings'
+
+-   Go to the 'Developer' tab in the side bar. Next to 'Access tokens', click the 'Manage' button
+
+::: {align="center"}
+![](../images/databricks-access-tokens.png)
+:::
+
+-   Click the 'Generate new token' button
+
+-   Name the token, then click 'Generate'
 
--   Access to Databricks
--   Access to a SQL Warehouse on DataBricks
--   R and RStudio downloaded
+::: callout-note
+Note that access tokens will only last as long as the value for the 'Lifetime (days)' field. After this period the token will expire, and you will need to create a new one to re-authenticate.
+:::
+
+-   Make a note of the 'Databricks access token' it has given you
+
+::: callout-warning
+It is very important that you immediately copy the access token that you are given, as you will not be able to see it through Databricks again. If you lose this access token before pasting it into RStudio then you must generate a new access token to replace it.
+:::
 
 ------------------------------------------------------------------------
 
-# Downloading an ODBC driver
+#### Set up ODBC connection from your laptop
```

Comment:
This section (the entire "Set up ODBC connection from your laptop" section) isn't necessary for connections between R and Databricks, because `odbc::databricks` handles all of it if it has the environment variables. Creating the odbc connection in `ODBC Data Sources (64-bit)` is only needed if you're connecting through Excel. We could just move all of this to a different section?

Reply:
Thanks - this is the part I wasn't sure about!